From ac15f60a6a1d8c6719e811dda8bcc42efbc661a7 Mon Sep 17 00:00:00 2001
From: Shalini Salomi Bodapati <shalini.salomi.bodapati@intel.com>
Date: Mon, 24 May 2021 13:14:08 +0530
Subject: [PATCH] Modify install hook check in art

The install hook can be called for more than one module variant for the
same library (for host). This patch modifies path len check so as to
avoid conflict for avx2 version of libs for x86 and x86_64

Change-Id: I720e1af122a185bd1419b408e40b8bd0f146b830
Tracked-On:
Signed-off-by: Shalini Salomi Bodapati <shalini.salomi.bodapati@intel.com>
---
 build/art.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/art.go b/build/art.go
index a24da800f21d..051341c6588f 100644
--- a/build/art.go
+++ b/build/art.go
@@ -316,8 +316,8 @@ func addTestcasesFile(ctx android.InstallHookContext) {
 
 	src := ctx.SrcPath().String()
 	path := strings.Split(ctx.Path().ToMakePath().String(), "/")
-	// Keep last two parts of the install path (e.g. bin/dex2oat).
-	dst := strings.Join(path[len(path)-2:], "/")
+	// Keep last four parts of the install path (e.g. bin/dex2oat).
+	dst := strings.Join(path[len(path)-4:], "/")
 	if oldSrc, ok := testcasesContent[dst]; ok {
 		ctx.ModuleErrorf("Conflicting sources for %s: %s and %s", dst, oldSrc, src)
 	}
-- 
2.17.1

